operator-wg/whitepaper-community-review/README.md

```diff
@@ -0,0 +1,882 @@
+of completeness, this chapter deals with some best practice when writing
+own Operators, based on our own experience or described by the
+community. However, without clear knowledge of the actual state and
+without clear ideas of what we want to achieve, we also need some
+methods and techniques specifying what our Operator should do.
+Therefore, we will also have to deal with some aspects of requirement
+engineering.
+
+### Requirement analysis
+
+A key promise of Kubernetes is that it enables the automation of operational
+tasks to deploy, scale, and manage containerized applications across
+multiple environments with no (or minimal) human intervention. In
+Kubernetes, stateless cloud-native applications are well suited for
+horizontal scaling, automated self-healing restarts, or progressive
+rollout of new containers. However, stateful applications with complex
+components running in clustered or distributed environments are not
+always well suited for this type of container-based infrastructure. They
+still require human interaction when it comes to persistence, upgrades,
+or high availability to remain in a stable state.
+
+True, Kubernetes solves these issues in a novel way by creating and
+managing custom applications using Operators. However, and here is the
+first question: as a developer, do you really know how this type of
+application works and interacts both internally and externally? How do
+the day-to-day IT operations work? How is the application backed up
+(including recovery)? What steps are necessary in case of failovers or
+outages, are there any dependencies between the software components?
+
+It is therefore strongly recommended that a comprehensive requirement
+analysis is needed to determine the requirements or conditions of an
+Operator. Requirement analysis is critical to the success or failure of
+Operators. All requirements should be documented, measurable, testable,
+traceable, related to identified requirements, and defined at a level of
+detail sufficient for system design.
+
+Best practice:
+
+1.  Do not assume to know what is needed.
+
+2.  Study existing documentation of your application, interview
+    responsible system administrators and other stakeholders (if
+    necessary), get a list of possible system check activities, Business
+    and SLA-relevant KPI and compare them with existing incident
+    reports or bug tracking lists.
+
+3.  Describe a concrete scenario (e.g., application failover) in detail
+    along the lines of "who does what, when, how, and why".
+
+4.  Describe what an Operator needs to know to run the previous scenario
+    independently, keeping the application in a stable and productive
+    state.
+
+### Custom or third-party Operator
+
+Now that we know that we need an Operator and what the requirements are,
+we can turn our attention to the question where do we get a Kubernetes
+Operator which meets best our requirements?
```

Comment:
```suggestion
Now that the situations where using an Operator have been made clear, the next part of the paper will focus on where Operator implementations are available and which best meets requirements.
```

I just felt this needed a statement rather than rhetorical questions.